k8s/cmd/admission-webhook/patches.go

```diff
@@ -0,0 +1,185 @@
+				VolumeMounts: []corev1.VolumeMount{{
+					ReadOnly:  false,
+					Name:      "nsm-coredns-volume",
+					MountPath: "/etc/coredns",
+				}},
+				Env: []corev1.EnvVar{
+					{
+						Name:  env.UseUpdateAPIEnv.Name(),
+						Value: "true",
+					},
+					{
+						Name:  env.UpdateAPIClientSock.Name(),
+						Value: "/etc/coredns/client.sock",
+					},
+				},
+			},
+		})...)
+	patch = append(patch, addContainer(tuple.spec,
+		[]corev1.Container{
+			{
+				Name:            "nsm-dns-monitor",
+				Image:           fmt.Sprintf("%s/%s:%s", getRepo(), "nsm-monitor", getTag()),
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				Env: []corev1.EnvVar{
+					{
+						Name:  "MONITOR_DNS_CONFIGS_ENV",
+						Value: "true",
+					},
+					{
+						Name:  env.UpdateAPIClientSock.Name(),
+						Value: "/etc/coredns/client.sock"},
+					{
+						Name:  client.AnnotationEnv,
+						Value: annotationValue,
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{{
+					ReadOnly:  false,
+					Name:      "nsm-coredns-volume",
+					MountPath: "/etc/coredns",
+				}},
+				Resources: corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"networkservicemesh.io/socket": resource.MustParse("1"),
+					},
+				},
+			},
+		})...)
+	patch = append(patch, addVolume(tuple.spec,
+		[]corev1.Volume{{
+			Name: "nsm-coredns-volume",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{
+					Medium:    corev1.StorageMediumDefault,
+					SizeLimit: nil,
+				},
+			},
+		}})...)
+	patch = append(patch, replaceDNSConfig()...)
+	return patch
+}
+
+func createNsmInitContainerPatch(annotationValue string) []patchOperation {
+	var patch []patchOperation
+
+	envVals := []corev1.EnvVar{{
+		Name:  client.AnnotationEnv,
+		Value: annotationValue,
+	}}
+	jaegerHost := getJaegerHost()
+	if jaegerHost != "" {
+		envVals = append(envVals,
+			corev1.EnvVar{
+				Name:  jaegerHostEnv,
+				Value: jaegerHost,
+			})
+	}
+	jaegerPort := getJaegerPort()
+	if jaegerPort != "" {
+		envVals = append(envVals,
+			corev1.EnvVar{
+				Name:  jaegerPortEnv,
+				Value: jaegerPort,
+			})
+	}
+
+	value := []corev1.Container{{
+		Name:            initContainerName,
+		Image:           fmt.Sprintf("%s/%s:%s", getRepo(), getInitContainer(), getTag()),
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Env:             envVals,
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				"networkservicemesh.io/socket": resource.MustParse("1"),
+			},
+		},
+	}}
+
+	patch = append(patch, patchOperation{
+		Op:    "add",
+		Path:  initContainersPath,
+		Value: value,
+	})
+
+	return patch
+}
+
+func replaceDNSConfig() []patchOperation {
+	return []patchOperation{{
+		Op:   "replace",
+		Path: dnsConfigPath,
+		Value: &corev1.PodDNSConfig{
+			Nameservers: []string{"127.0.0.1"},
+			Searches:    []string{"default.svc.cluster.local", "svc.cluster.local", "cluster.local"},
```

Comment:
How we will handle changes in this defaults, is it possible to retrieve this server names from cluster?

Reply:
Unfortunately, we can't reuse staff from k8s, please take a look at https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/dns/dns.go#L74 it super dependes on kubelet specific things.

Added env for posible to cofigure dns search names as we discussed.